sqlx-sqlite/src/connection/mod.rs

```diff
@@ -112,6 +115,50 @@ impl SqliteConnection {
 
         Ok(LockedSqliteHandle { guard })
     }
+
+    /// Serializes the SQLite database to an in-memory buffer.
+    ///
+    /// This function allows you to serialize an SQLite database to an in-memory buffer.
+    /// The output is the same bytes as if the database were to be written to disk.
+    pub async fn serialize_to_buffer(&mut self) -> Result<SerializedDatabase, Error> {
+        let locked = self.lock_handle().await?;
+
+        unsafe {
+            let mut size: sqlite3_int64 = 0;
+            let data_ptr = sqlite3_serialize(
```

Comment:
This may return a null pointer if a memory allocation error occurs: https://www.sqlite.org/c3ref/serialize.html